pkg/hosts/privdrop_unix_test.go

```
@@ -0,0 +1,236 @@
+			defer restore()
+
+			err := applyPrivilegeDrop(501, 20)
+			require.Error(t, err)
+			assert.Contains(t, err.Error(), tt.wantErr)
+		})
+	}
+}
+
+func TestOpenHostsFileAndDropPrivilegesClosesFileOnDropFailure(t *testing.T) {
+	hostsFile := filepath.Join(t.TempDir(), "hosts")
+	require.NoError(t, os.WriteFile(hostsFile, []byte("keep"), 0644))
```

golangci-lint (2.12.2)

[medium] G306: Expect WriteFile permissions to be 0600 or less

(gosec)